test/e2e/providers/vsphere/vsphere.go

```diff
@@ -52,19 +66,44 @@ func newVSphereMachineProviderSpec(clusterID string) (*vsphere.VSphereMachinePro
 		},
 		NumCPUs:           int32(4),
 		NumCoresPerSocket: int32(1),
-		// The template is hardcoded with an image which has been properly sysprepped.
-		// TODO: Find a way to automatically update this with latest image
-		Template: "windows-golden-images/windows-server-2004-template",
-		Workspace: &vsphere.Workspace{
-			Datacenter:   "SDDC-Datacenter",
-			Datastore:    "WorkloadDatastore",
-			Folder:       "/SDDC-Datacenter/vm/" + clusterID,
-			ResourcePool: "/SDDC-Datacenter/host/Cluster-1/Resources",
-			Server:       "vcenter.sddc-44-236-21-251.vmwarevmc.com",
-		},
+
+		Template:  vmTemplate,
+		Workspace: workspace,
 	}, nil
 }
 
+// getWorkspaceFromExistingMachineSet returns Workspace from a machineset provisioned during installation
+func (p *Provider) getWorkspaceFromExistingMachineSet() (*vsphere.Workspace, error) {
+	clusterID, err := p.oc.GetInfrastructureID()
+	if err != nil {
+		return nil, fmt.Errorf("unable to get cluster id: %v", err)
+	}
+
+	listOptions := metav1.ListOptions{LabelSelector: "machine.openshift.io/cluster-api-cluster=" + clusterID}
+	machineSets, err := p.oc.Machine.MachineSets("openshift-machine-api").List(context.TODO(), listOptions)
+	if err != nil {
+		return nil, fmt.Errorf("unable to get machinesets: %v", err)
+	}
+
+	if len(machineSets.Items) == 0 {
+		return nil, fmt.Errorf("no matching machinesets found")
+	}
+
+	machineSet := machineSets.Items[0]
+	templateSpec := machineSet.Spec.Template.Spec
+	providerSpecRaw := templateSpec.ProviderSpec.Value
```

Comment:
This can be simplified to `providerSpecRaw := machineSet.Spec.Template.Spec.ProviderSpec.Value`
Having the extra variable `templateSpec` is confusing as we are only interested in the ProviderSpec's value.